File: backend/src/infrastructure/persistence/models/catalog.py
from django.db import models

from infrastructure.persistence.models.user import Concesionaria
from .base import BaseModel


class CatalogParamType(models.TextChoices):
    VEHICLE_MODEL = "VEHICLE_MODEL", "Vehicle Model"
    COLOR = "COLOR", "Color"
    WARRANTY_TYPE = "WARRANTY_TYPE", "Warranty Type"
    CAMPAIGN_CODE = "CAMPAIGN_CODE", "Campaign Code"


class CatalogParam(BaseModel):
    param_type = models.CharField(max_length=20, choices=CatalogParamType.choices)
    code = models.CharField(max_length=50)
    name = models.CharField(max_length=200)
    concesionaria = models.CharField(max_length=20, choices=Concesionaria.choices)

    class Meta:
        db_table = "catalog_params"
        unique_together = [("param_type", "code", "concesionaria")]
        ordering = ["param_type", "name"]

    def __str__(self) -> str:
        return f"[{self.param_type}] {self.code} — {self.name}"


class SparePart(BaseModel):
    catalog_code = models.CharField(max_length=20)
    name = models.CharField(max_length=200)
    unit_price = models.DecimalField(max_digits=10, decimal_places=2)
    concesionaria = models.CharField(max_length=20, choices=Concesionaria.choices)

    class Meta:
        db_table = "spare_parts"
        unique_together = [("catalog_code", "concesionaria")]
        ordering = ["name"]

    def __str__(self) -> str:
        return f"{self.catalog_code} — {self.name}"

File: frontend/src/types/catalog.ts
export type CatalogParamType =
  | "VEHICLE_MODEL"
  | "COLOR"
  | "WARRANTY_TYPE"
  | "CAMPAIGN_CODE";

export const CATALOG_PARAM_LABELS: Record<CatalogParamType, string> = {
  VEHICLE_MODEL: "Modelos de Vehículo",
  COLOR:         "Colores",
  WARRANTY_TYPE: "Tipos de Garantía",
  CAMPAIGN_CODE: "Códigos de Campaña",
};

export interface CatalogParam {
  id:           string;
  param_type:   string;
  code:         string;
  name:         string;
  concesionaria: string;
}

export interface SparePart {
  id:           string;
  catalog_code: string;
  name:         string;
  unit_price:   string;
  concesionaria: string;
}
